OpenRA.Game/Player.cs

```diff
@@ -161,16 +163,24 @@ public Player(World world, Session.Client client, PlayerReference pr)
 				DisplayFaction = ChooseDisplayFaction(world, pr.Faction);
 			}
 
+			IsBot = BotType != null;
+
 			if (!Spectating)
 				PlayerMask = new LongBitSet<PlayerBitMask>(InternalName);
 
-			var playerActorType = world.Type == WorldType.Editor ? "EditorPlayer" : "Player";
-			PlayerActor = world.CreateActor(playerActorType, new TypeDictionary { new OwnerInit(this) });
+			// Special case handling is required for the Player actor:
+			// Since Actor.Created would be called before PlayerActor is assigned here
+			// querying player traits in INotifyCreated.Created would crash.
+			// Therefore only call the constructor of Actor and run the Created callbacks
```

Comment:
Maybe `// Therefore assign the uninitialized actor and run the Created callbacks` would be more explicit?